src/Promitor.Agents.Scraper/Scheduling/ResourcesScrapingJob.cs

```diff
@@ -287,11 +287,11 @@ private async Task ScrapeMetricBatched(BatchScrapeDefinition<IAzureResourceDefin
                 var resourceSubscriptionId = batchScrapeDefinition.ScrapeDefinitionBatchProperties.SubscriptionId;
                 var azureMonitorClient = _azureMonitorClientFactory.CreateIfNotExists(_metricsDeclaration.AzureMetadata.Cloud, _metricsDeclaration.AzureMetadata.TenantId,
                     resourceSubscriptionId, _metricSinkWriter, _azureScrapingSystemMetricsPublisher, _resourceMetricDefinitionMemoryCache, _configuration,
-                    _azureMonitorIntegrationConfiguration, _azureMonitorLoggingConfiguration, _loggerFactory);
-                var azureEnvironent = _metricsDeclaration.AzureMetadata.Cloud.GetAzureEnvironment();
+                    _azureMonitorIntegrationConfiguration, _azureMonitorLoggingConfiguration, _loggerFactory, _metricsDeclaration.AzureMetadata.Endpoints);
+                var azureEnvironent = _metricsDeclaration.AzureMetadata.Cloud.GetAzureEnvironment(_metricsDeclaration.AzureMetadata.Endpoints);
```

Comment:
Instead of passing this everywhere, we should change it to something like this:
```suggestion
                var azureEnvironent = _metricsDeclaration.AzureMetadata.GetAzureEnvironment();
```

Reply:
I've updated the code accordingly, except for the calls made to methods in `Promitor.Integrations.AzureMonitor`. These methods currently do not reference `Promitor.Core.Scraping.Configuration.Model.AzureMetadata`, and adding this reference would introduce a circular dependency, as the flow of dependencies currently goes from `Promitor.Core.Scraping` to `Promitor.Integrations.AzureMonitor`.